metadata/src/test/java/org/apache/kafka/controller/QuorumControllerTest.java

```diff
@@ -654,6 +658,19 @@ public void testMinIsrUpdateWithElr() throws Throwable {
 
             // Unfence all brokers and create a topic foo (min ISR 2)
             sendBrokerHeartbeatToUnfenceBrokers(active, allBrokers, brokerEpochs);
+            AtomicBoolean onlyKeepUnfenced = new AtomicBoolean(false);
+            exec.scheduleAtFixedRate(() -> {
```

Comment:
What was the underlying problem with sending the heartbeat sync'ly throughout the test? Was it a timing issue or some kind of race?